lib/edge-runtime/requests.ts

```diff
@@ -165,20 +169,33 @@ export function constructHttpRequest({
       url += `?${queryParams.toString()}`;
     }
   }
-  const logMessage = `Attempting fetch with url: ${url}\n\nWith options:${JSON.stringify(
-    requestOptions,
-    null,
-    2
-  )}`;
-  console.log(logMessage);
+  return url;
+}
 
-  if (stream)
-    stream({
-      role: "debug",
-      content: logMessage,
-    });
+function buildBody(
```

Comment:
i was getting brain damage trying to follow all of the logic in createHttpRequest

Reply:
I see your point

However, I think in this case, it's easier to read as a long function. Imo certainly the `buildUrl()` function shouldn't be extracted since it has the discussed side effect going on.

Shall we compromise by returning `buildUrl()` back to its previous state (with the `requestOptions.headers` change made) and leave `buildBody()`?

Think [this discussion](https://www.reddit.com/r/ExperiencedDevs/comments/141ild0/are_you_in_favor_of_small_functionsclean_code_or/) is worth a quick read on the topic